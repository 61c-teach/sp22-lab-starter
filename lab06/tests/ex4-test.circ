<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<project source="3.4.3-61c.cc0f4a6" version="1.0">
  This file is intended to be loaded by Logisim-evolution (https://github.com/reds-heig/logisim-evolution).

  <lib desc="#Wiring" name="0">
    <tool name="Splitter">
      <a name="appear" val="center"/>
      <a name="fanout" val="4"/>
      <a name="incoming" val="4"/>
    </tool>
    <tool name="Pin">
      <a name="facing" val="north"/>
    </tool>
    <tool name="Probe">
      <a name="radix" val="16"/>
    </tool>
    <tool name="Tunnel">
      <a name="width" val="2"/>
    </tool>
    <tool name="Pull Resistor">
      <a name="facing" val="north"/>
    </tool>
    <tool name="Clock">
      <a name="facing" val="north"/>
    </tool>
    <tool name="Constant">
      <a name="value" val="0xff"/>
      <a name="width" val="8"/>
    </tool>
  </lib>
  <lib desc="#Gates" name="1">
    <tool name="PLA">
      <a name="table" val=""/>
    </tool>
  </lib>
  <lib desc="#Plexers" name="2">
    <tool name="Multiplexer">
      <a name="enable" val="false"/>
    </tool>
    <tool name="Demultiplexer">
      <a name="enable" val="false"/>
    </tool>
  </lib>
  <lib desc="#Arithmetic" name="3"/>
  <lib desc="#Memory" name="4">
    <tool name="D Flip-Flop">
      <a name="appearance" val="logisim_evolution"/>
    </tool>
    <tool name="T Flip-Flop">
      <a name="appearance" val="logisim_evolution"/>
    </tool>
    <tool name="J-K Flip-Flop">
      <a name="appearance" val="logisim_evolution"/>
    </tool>
    <tool name="S-R Flip-Flop">
      <a name="appearance" val="logisim_evolution"/>
    </tool>
    <tool name="Register">
      <a name="appearance" val="logisim_evolution"/>
    </tool>
    <tool name="Counter">
      <a name="appearance" val="logisim_evolution"/>
    </tool>
    <tool name="Shift Register">
      <a name="appearance" val="logisim_evolution"/>
    </tool>
    <tool name="Random">
      <a name="appearance" val="logisim_evolution"/>
    </tool>
    <tool name="RAM">
      <a name="appearance" val="logisim_evolution"/>
    </tool>
    <tool name="ROM">
      <a name="appearance" val="logisim_evolution"/>
      <a name="contents">addr/data: 8 8
0
</a>
    </tool>
  </lib>
  <lib desc="#Base" name="5">
    <tool name="Text Tool">
      <a name="font" val="SansSerif plain 12"/>
      <a name="halign" val="center"/>
      <a name="text" val=""/>
      <a name="valign" val="base"/>
    </tool>
  </lib>
  <lib desc="file#../ex4.circ" name="6"/>
  <main name="main"/>
  <options>
    <a name="gateUndefined" val="ignore"/>
    <a name="simlimit" val="1000"/>
    <a name="simrand" val="0"/>
  </options>
  <mappings>
    <tool lib="5" map="Button2" name="Menu Tool"/>
    <tool lib="5" map="Button3" name="Menu Tool"/>
    <tool lib="5" map="Ctrl Button1" name="Menu Tool"/>
  </mappings>
  <toolbar>
    <tool lib="5" name="Poke Tool"/>
    <tool lib="5" name="Edit Tool"/>
    <tool lib="5" name="Wiring Tool"/>
    <tool lib="5" name="Text Tool">
      <a name="font" val="SansSerif plain 12"/>
      <a name="halign" val="center"/>
      <a name="text" val=""/>
      <a name="valign" val="base"/>
    </tool>
    <sep/>
    <tool lib="0" name="Pin">
      <a name="appearance" val="NewPins"/>
    </tool>
    <tool lib="0" name="Pin">
      <a name="appearance" val="NewPins"/>
      <a name="facing" val="west"/>
      <a name="output" val="true"/>
    </tool>
    <sep/>
    <tool lib="1" name="NOT Gate"/>
    <tool lib="1" name="AND Gate"/>
    <tool lib="1" name="OR Gate"/>
    <tool lib="1" name="XOR Gate"/>
    <tool lib="1" name="NAND Gate"/>
    <tool lib="1" name="NOR Gate"/>
    <sep/>
    <tool lib="4" name="D Flip-Flop">
      <a name="appearance" val="logisim_evolution"/>
    </tool>
    <tool lib="4" name="Register">
      <a name="appearance" val="logisim_evolution"/>
    </tool>
  </toolbar>
  <circuit name="main">
    <a name="appearance" val="logisim_evolution"/>
    <a name="circuit" val="main"/>
    <a name="circuitnamedboxfixedsize" val="true"/>
    <a name="circuitvhdlpath" val=""/>
    <a name="clabel" val=""/>
    <a name="clabelfont" val="SansSerif bold 16"/>
    <a name="clabelup" val="east"/>
    <comp lib="0" loc="(110,140)" name="Tunnel">
      <a name="facing" val="east"/>
      <a name="label" val="CLK"/>
    </comp>
    <comp lib="0" loc="(110,420)" name="Clock">
      <a name="label" val="CLK"/>
    </comp>
    <comp lib="0" loc="(140,420)" name="Tunnel">
      <a name="label" val="CLK"/>
    </comp>
    <comp lib="0" loc="(160,50)" name="Pin">
      <a name="appearance" val="NewPins"/>
      <a name="facing" val="west"/>
      <a name="label" val="halt"/>
      <a name="locked" val="true"/>
      <a name="output" val="true"/>
    </comp>
    <comp lib="0" loc="(200,100)" name="Pin">
      <a name="appearance" val="NewPins"/>
      <a name="facing" val="west"/>
      <a name="label" val="CYCLE"/>
      <a name="locked" val="true"/>
      <a name="output" val="true"/>
      <a name="width" val="4"/>
    </comp>
    <comp lib="0" loc="(450,130)" name="Tunnel">
      <a name="facing" val="east"/>
      <a name="label" val="INPUT1"/>
      <a name="width" val="8"/>
    </comp>
    <comp lib="0" loc="(450,170)" name="Tunnel">
      <a name="facing" val="east"/>
      <a name="label" val="INPUT2"/>
      <a name="width" val="8"/>
    </comp>
    <comp lib="0" loc="(450,210)" name="Tunnel">
      <a name="facing" val="east"/>
      <a name="label" val="CLK"/>
    </comp>
    <comp lib="0" loc="(50,110)" name="Constant">
      <a name="width" val="4"/>
    </comp>
    <comp lib="0" loc="(60,40)" name="Constant">
      <a name="value" val="0xa"/>
      <a name="width" val="4"/>
    </comp>
    <comp lib="0" loc="(620,350)" name="Tunnel">
      <a name="label" val="INPUT1"/>
      <a name="width" val="8"/>
    </comp>
    <comp lib="0" loc="(640,600)" name="Tunnel">
      <a name="label" val="INPUT2"/>
      <a name="width" val="8"/>
    </comp>
    <comp lib="0" loc="(750,130)" name="Pin">
      <a name="appearance" val="NewPins"/>
      <a name="facing" val="west"/>
      <a name="label" val="OUT1"/>
      <a name="locked" val="true"/>
      <a name="output" val="true"/>
      <a name="width" val="8"/>
    </comp>
    <comp lib="0" loc="(830,170)" name="Tunnel">
      <a name="facing" val="east"/>
      <a name="label" val="INPUT1"/>
      <a name="width" val="8"/>
    </comp>
    <comp lib="0" loc="(830,210)" name="Tunnel">
      <a name="facing" val="east"/>
      <a name="label" val="INPUT2"/>
      <a name="width" val="8"/>
    </comp>
    <comp lib="0" loc="(880,170)" name="Pin">
      <a name="appearance" val="NewPins"/>
      <a name="facing" val="west"/>
      <a name="label" val="IN1"/>
      <a name="locked" val="true"/>
      <a name="output" val="true"/>
      <a name="width" val="8"/>
    </comp>
    <comp lib="0" loc="(880,210)" name="Pin">
      <a name="appearance" val="NewPins"/>
      <a name="facing" val="west"/>
      <a name="label" val="IN2"/>
      <a name="locked" val="true"/>
      <a name="output" val="true"/>
      <a name="width" val="8"/>
    </comp>
    <comp lib="3" loc="(100,100)" name="Adder">
      <a name="width" val="4"/>
    </comp>
    <comp lib="3" loc="(110,50)" name="Comparator">
      <a name="mode" val="unsigned"/>
      <a name="width" val="4"/>
    </comp>
    <comp lib="4" loc="(110,70)" name="Register">
      <a name="appearance" val="logisim_evolution"/>
      <a name="width" val="4"/>
    </comp>
    <comp lib="4" loc="(360,290)" name="ROM">
      <a name="addrWidth" val="4"/>
      <a name="appearance" val="logisim_evolution"/>
      <a name="contents">addr/data: 4 8
3 2 fc 1 2 fb 1
</a>
      <a name="labelvisible" val="true"/>
    </comp>
    <comp lib="4" loc="(370,540)" name="ROM">
      <a name="addrWidth" val="4"/>
      <a name="appearance" val="logisim_evolution"/>
      <a name="contents">addr/data: 4 8
4 5 70 9 d c 2
</a>
    </comp>
    <comp lib="6" loc="(720,130)" name="pipelined"/>
    <wire from="(100,100)" to="(110,100)"/>
    <wire from="(110,420)" to="(140,420)"/>
    <wire from="(110,50)" to="(160,50)"/>
    <wire from="(170,100)" to="(190,100)"/>
    <wire from="(190,100)" to="(190,180)"/>
    <wire from="(190,100)" to="(200,100)"/>
    <wire from="(190,180)" to="(350,180)"/>
    <wire from="(30,180)" to="(190,180)"/>
    <wire from="(30,60)" to="(30,90)"/>
    <wire from="(30,60)" to="(70,60)"/>
    <wire from="(30,90)" to="(30,180)"/>
    <wire from="(30,90)" to="(60,90)"/>
    <wire from="(350,180)" to="(350,300)"/>
    <wire from="(350,300)" to="(350,550)"/>
    <wire from="(350,300)" to="(360,300)"/>
    <wire from="(350,550)" to="(370,550)"/>
    <wire from="(450,130)" to="(500,130)"/>
    <wire from="(450,170)" to="(460,170)"/>
    <wire from="(450,210)" to="(500,210)"/>
    <wire from="(460,150)" to="(460,170)"/>
    <wire from="(460,150)" to="(500,150)"/>
    <wire from="(50,110)" to="(60,110)"/>
    <wire from="(500,170)" to="(500,210)"/>
    <wire from="(60,40)" to="(70,40)"/>
    <wire from="(600,350)" to="(620,350)"/>
    <wire from="(610,600)" to="(640,600)"/>
    <wire from="(720,130)" to="(750,130)"/>
    <wire from="(830,170)" to="(880,170)"/>
    <wire from="(830,210)" to="(880,210)"/>
  </circuit>
</project>
