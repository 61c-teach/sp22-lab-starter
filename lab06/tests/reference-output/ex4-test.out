CYCLE,OUT1,IN1,IN2
0000,00000000,00000011,00000100
0001,00000000,00000010,00000101
0010,00001100,11111100,01110000
0011,00010110,00000001,00001001
0100,01010110,00000010,00001101
0101,01011111,11111011,00001100
0110,01111001,00000001,00000010
0111,00111101,00000000,00000000
1000,00111111,00000000,00000000
1001,00111111,00000000,00000000
1010,00111111,00000000,00000000
